kratos/utilities/math_utils.h

```diff
@@ -1539,6 +1547,277 @@ class MathUtils
         return is_converged;
     }
      
+    /**
+     * This function gives the Jacobi SVD of a given mxn matrix (m>=n), returns U,S; where A=U*S*V
+     * U and V are unitary, and S is a diagonal matrix. 
+     * Where s_i >= 0, and s_i >= s_i+1 (which means that the biggest number is the first one and the smallest the last one)
+     * TODO: This version is quite innefficient, look for a real and mathematical implementation (not the algorithm found in Wikipedia!!)
+     * @param InputMatrix: The matrix where perform the SVD 
+     * @param UMatrix: The unitary U matrix
+     * @param SMatrix: The diagonal S matrix
+     * @param VMatrix: The unitary V matrix
+     * @param Tolerance: The tolerance considered
+     * @return iter: The number of iterations
+     */
+     
+    static inline std::size_t SingularValueDecomposition(
+        const MatrixType& InputMatrix,
+        MatrixType& UMatrix,
+        MatrixType& SMatrix,
+        MatrixType& VMatrix,
+        const TDataType Tolerance = std::numeric_limits<double>::epsilon()
+        )
+    {
+        const SizeType m = InputMatrix.size1();
+        const SizeType n = InputMatrix.size2();
+        
+        SMatrix.resize(m, n);
+        SMatrix.assign(InputMatrix);
+//         SMatrix = InputMatrix;
+        
+        UMatrix.resize(m, m);
+        noalias(UMatrix) = IdentityMatrix(m, m);
+        
+        VMatrix.resize(n, n);
+        noalias(VMatrix) = IdentityMatrix(n, n);
+        
+        const TDataType relative_tolerance = Tolerance * SparseSpaceType::TwoNorm(InputMatrix);
+        
+        std::size_t iter = 0;
+        
+        while (SparseSpaceType::JacobiNorm(SMatrix) > relative_tolerance)
+        {
+            for (SizeType i = 0; i < n; i++)
+            {
+                for (SizeType j = i+1; j < n; j++)
+                {
+                    MatrixType j1(m, m);
+                    MatrixType j2(n, n);
+                    
+                    Jacobi(j1, j2, SMatrix, m, n, i, j);
+                    
+                    SMatrix = prod(j1, MatrixType(prod(SMatrix, j2)));
```

Comment:
Create explicitly a temp and use noakias in the second

Reply:
OK